### PR TITLE
Local session

### PR DESCRIPTION
**Pull request by sleidig**
see issue: #601

### Visible/Frontend Changes
_none_

### Architectural/Backend Changes
- [x] add an additional session mode "local" `NewLocalSessionService` that uses only a local PouchDB instance and no remote sync counterpart
- [x] clean up appConfig (config.json): separately set demo mode and session mode
- [x] tweak demo data generation to also work for the new local session mode

! If you have a local config.json file (and are not only using config.default.json) you need to update that file.

### CODE REVIEW

**Review by TheSlimvReal — APPROVED** [review]
Looks very good, just some general questions regarding the implementation.

File: src/app/core/session/session-service/new-local-session.service.ts [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+      throw error; // rethrow, so later Promise-handling lands in .catch, too
+    }
+  }
+
+  /**
+   * Get the local database instance that should be used for regular data access.
+   * als see {@link SessionService}
+   */
+  public getDatabase(): Database {
+    return new PouchDatabase(this.database, this.loggingService);
```
[the remark this answers is not in the document]
Comment (sleidig) [reply src/app/core/session/session-service/new-local-session.service.ts:173]: good point, took that from the existing local-session but briefly wondered about it, too. Changing this here

File: src/app/core/session/session-service/new-local-session.service.ts [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+import { SyncState } from "../session-states/sync-state.enum";
+import { User } from "../../user/user";
+import { EntitySchemaService } from "../../entity/schema/entity-schema.service";
+import { LoggingService } from "../../logging/logging.service";
+import { StateHandler } from "../session-states/state-handler";
+import PouchDB from "pouchdb-browser";
+import { AppConfig } from "../../app-config/app-config";
+
+@Injectable()
+export class NewLocalSessionService extends SessionService {
```
[the remark this answers is not in the document]
Comment (sleidig) [reply src/app/core/session/session-service/new-local-session.service.ts:34]: there already is a "LocalSession" :see_no_evil: 
Unfortunately that has some peculiarities in how it is intertwined with the SyncedSessionService. In the long term I would like to also use this to replace the old local session used in the synced session.

Creating an issue to refactor that further: #701

File: src/app/core/session/session-service/new-local-session.service.ts [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+  public getConnectionState() {
+    return this.connectionState;
+  }
+  /** see {@link SessionService} */
+  public getSyncState() {
+    return this.syncState;
+  }
+
+  /** see {@link SessionService} */
+  public async sync(remoteDatabase?): Promise<any> {
```
[the remark this answers is not in the document]
Comment (sleidig) [reply src/app/core/session/session-service/new-local-session.service.ts:154]: agreed. I am not sure why this was implemented this way. I listed this in the issue for further refactoring #701  but didn't want to change to overall interface at this point.

**Pull request merged**